front/components/assistant/conversation/ConversationViewer.tsx

```diff
@@ -449,3 +466,36 @@ const groupMessages = (
 
   return groups;
 };
+
+/**
+ * Organizes and categorizes groups of messages by their type.
+ *
+ * This function takes an array of message groups, each containing one or more messages,
+ * and groups these message groups into larger collections based on the type of the first
+ * message in each group. The function ensures that all message groups in each collection
+ * share the same type, facilitating type-specific processing or display.
+ */
+function groupMessagesByType(
+  groupedMessages: MessageWithContentFragmentsType[][]
+): MessageWithContentFragmentsType[][][] {
+  return groupedMessages.reduce<MessageWithContentFragmentsType[][][]>(
+    (typedGroupsAcc, currentMessage, index) => {
+      const isLastMessage = index === groupedMessages.length - 1;
+      const currentMessageType = currentMessage[0].type;
+      const lastGroup = typedGroupsAcc[typedGroupsAcc.length - 1];
+      const lastGroupType = lastGroup?.[0]?.[0]?.type;
+
+      if (!lastGroup || currentMessageType !== lastGroupType) {
+        typedGroupsAcc.push([currentMessage]);
+      } else {
+        lastGroup.push(currentMessage);
+      }
+
+      if (isLastMessage && currentMessageType === "user_message") {
+        typedGroupsAcc.push([]);
```

Comment:
Not sure to understand why we do this 🤔 ?

Reply:
Changed the logic, we don't need it anymore